aya/src/programs/links.rs

```diff
@@ -369,6 +369,180 @@ pub enum LinkError {
     SyscallError(#[from] SyscallError),
 }
 
+/// A [`Link`] identifier, which may or may not be owned by aya.
+pub struct LinkId(u32);
+
+impl LinkId {
+    /// A wrapper for an arbitrary loaded link specified by its kernel id,
+    /// unsafe because there is no guarantee provided by aya that the link is
+    /// still loaded or even exists.
+    pub unsafe fn new(id: u32) -> Self {
+        Self(id)
+    }
+}
+
+bitflags::bitflags! {
+    /// Flags which are use to build a set of MprogOptions.
+    #[derive(Clone, Copy, Debug, Default)]
+    pub(crate) struct MprogFlags: u32 {
+        const REPLACE = BPF_F_REPLACE;
```

Comment:
Is there a plan to support REPLACE?

Reply:
I can make a follow up issue to support Replace operations or add it here, It shouldn't be too much work

The harder part is adding tests for everything